core/src/main/scala/chisel3/experimental/verification/package.scala

```diff
@@ -9,16 +9,10 @@ import chisel3.internal.sourceinfo.SourceInfo
 
 package object verification {
 
-  /** Named class for assertions. */
-  final class Assert(val predicate: Bool) extends BaseSim
-
-  /** Named class for assumes. */
-  final class Assume(val predicate: Bool) extends BaseSim
-
-  /** Named class for covers. */
-  final class Cover(val predicate: Bool) extends BaseSim
-
   object assert {
+    /** Named class for assertions. */
+    final class Assert(val predicate: Bool) extends BaseSim
```

Comment:
```suggestion
    final class Assert extends BaseSim
```
Or
```scala
    final class(private[chisel3] val predicate: Bool) Assert extends BaseSim
```

I can't remember if we talked about this previously (I couldn't find any previous PR comments about this), but at the moment I'm leaning against publicly exposing the `predicate`, at least for now. My concern is about what behavior a user might expect vs. what they'll see. For example:

```scala
var assertRef: assert.Assert = _
when (foo) {
  assertRef = assert(fizz =/= buzz, "<useful message>")
}
// Use assertRef.predicate
```
My worry is that the users will expect `assertRef.predicate` to be equal to `!reset || !foo || (fizz =/= buzz)` rather than just `fizz =/= buzz`. I will grant that it could be useful though so I'm open to being convinced.

I'm curious what @ekiwi thinks about this API.

Perhaps if users want what I'm assuming they will expect, they can also capture `when.cond` so the onus is just on them there. For the record, I don't think we _want_ to capture `!reset || !foo || (fizz =/= buzz)` because it will either always generate FIRRTL nodes that are almost always unused or will be somewhat hard to get right (requires some extension of https://github.com/chipsalliance/chisel3/pull/1518)

Reply:
I think you bring up an interesting point. For now I think the main goal is to get annotation support into the `3.5` release, so I would keep `predicate` private. When you come up with a use-case for accessing predicate at SiFive, then you can always make another PR that just exposes predicate along with the user-story.